Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -47,10 +61,11 @@ class OpenSearchEngine {
 
     private func getURLFromTemplate(searchTemplate: String, query: String) -> NSURL? {
         let allowedCharacters = NSCharacterSet(charactersInString: SearchTermsAllowedCharacters)
-
         if let escapedQuery = query.stringByAddingPercentEncodingWithAllowedCharacters(allowedCharacters) {
             let urlString = searchTemplate.stringByReplacingOccurrencesOfString("{searchTerms}", withString: escapedQuery, options: NSStringCompareOptions.LiteralSearch, range: nil)
-            return NSURL(string: urlString)
+            if let encodedUrlString = urlString.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.symbollessURLAllowedSet()) {
```

Comment:
Yeah, but what if we omit that first line and use all the others? I imagine all the other URL character sets would also filter out invalid symbols/characters since they must be URL-compatible. Testing this in a playground, this seems to work still:
```
characterSet = NSMutableCharacterSet()
characterSet.formUnionWithCharacterSet(NSCharacterSet.URLQueryAllowedCharacterSet())
characterSet.formUnionWithCharacterSet(NSCharacterSet.URLUserAllowedCharacterSet())
characterSet.formUnionWithCharacterSet(NSCharacterSet.URLPathAllowedCharacterSet())
characterSet.formUnionWithCharacterSet(NSCharacterSet.URLPasswordAllowedCharacterSet())
characterSet.formUnionWithCharacterSet(NSCharacterSet.URLHostAllowedCharacterSet())
characterSet.formUnionWithCharacterSet(NSCharacterSet.URLFragmentAllowedCharacterSet())

let url = "https://ja.m.wikipedia.org/wiki/特?abc=123&foo=bar".stringByAddingPercentEncodingWithAllowedCharacters(characterSet)
```